components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/util/CaptchaUtil.java

```
@@ -616,6 +622,13 @@ private static void setReCaptchaConfigs(Properties properties) {
             throw new RuntimeException(getValidationErrorMessage(CaptchaConstants.RE_CAPTCHA_SCORE_THRESHOLD));
         }
 
+        try {
+            Double reCaptchaWarnScoreThreshold = getReCaptchaWarnThreshold(properties);
+            CaptchaDataHolder.getInstance().setReCaptchaWarnScoreThreshold(reCaptchaWarnScoreThreshold);
+        } catch (NumberFormatException e) {
+            throw new RuntimeException(getValidationErrorMessage(CaptchaConstants.RE_CAPTCHA_SCORE_THRESHOLD));
```

Exception is hidden here. Check how we handle exceptions in other places

---

Exception handling has been done similar to recaptcha score threshold https://github.com/wso2-extensions/identity-governance/blob/master/components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/util/CaptchaUtil.java#L612-L617

---

IMO i dont think its a good practice to break the flow due to this kind of invalid input.Its better if we can handle it set to default put a warn log and proceed

---

@hwupathum wdyt?

---

If there is a default value, +1 we don't need to break the flow

---

Done with [7c693bd](https://github.com/wso2-extensions/identity-governance/pull/768/commits/7c693bda8d996fc7b4fb5fb7f201709b840aaa37)